pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -3382,6 +3382,18 @@ func (la *LogicalAggregation) tryToGetMppHashAggs(prop *property.PhysicalPropert
 			}
 		}
 	}
+	// ref: https://github.com/pingcap/tiflash/blob/3ebb102fba17dce3d990d824a9df93d93f1ab
+	// 766/dbms/src/Flash/Coprocessor/AggregationInterpreterHelper.cpp#L26
+	validMppAgg := func(mppAgg *PhysicalHashAgg) bool {
+		isFinalOrCompleteMode := true
+		for _, one := range mppAgg.AggFuncs {
+			if one.Mode == aggregation.FinalMode || one.Mode == aggregation.CompleteMode {
+				continue
+			}
+			isFinalOrCompleteMode = false
+		}
+		return isFinalOrCompleteMode
```

Comment:
```suggestion
		for _, one := range mppAgg.AggFuncs {
			if one.Mode != aggregation.FinalMode && one.Mode != aggregation.CompleteMode {
				return false
			}
		}
		return true
```
I think this would be simpler and has the same effect.